src/lib.rs

```diff
@@ -1,3 +1,13 @@
+#![deny(warnings)]
+#![deny(unused)]
+#![deny(future_incompatible)]
+#![deny(bad_style)]
+#![warn(missing_docs)]
```

Comment:
As mentioned in a comment, I want to refrain from including `warn(missing_docs)` as well as the others below that for now. (I'm not sure where `box_pointers` is coming from, or why we would need it.)

Reply:
Sorry, will fix that.

Reply:
`box_pointers` is a lint that prevents use of owned heap memory. This restriction seems unneeded for this library, but it exists as a lint. Due to a bug in `rustc` it is not yet included in `deny(warnings)`. I've removed the exception for now.

Reply:
Do understand that adding the crate-wide Clippy attributes does no harm. It requires no extra dependencies, for example. The attributes are only effective when one installs `cargo-clippy` as a standalone tool and runs it on the project. Since there is no CI yet, the attributes have no effect on you should you choose to write new code that Clippy does not like. By adding those attributes you do give guidance when Clippy is used by a contributor in the future. Not having the attributes means that as I write new code, I'd have to put in more effort to lint it before filing a PR. Using Clippy in this way, with attributes in the code, is the best practice. Could you reconsider this please?

Reply:
As far as I can tell, the only effect of these attributes is whether Clippy treats things it finds as warning or error. I don't really understand how that would impede your work -- you can lint them just as warnings, I'd say. Certainly the way I run Clippy I prefer to get warnings rather than errors.